src/api/profile.ts

```diff
@@ -38,3 +39,37 @@ export async function deleteProfile(opts?: AxiosRequestConfig) {
 
   return deleteOK;
 }
+
+/**
+ * Function to get customer account ID based on phone number
+ *
+ * @param phoneNumber phone number with prefix
+ * @returns {string | GetAccountByPhoneErrorCode}
+ *  - customer_account_id as string if successful,
+ *  - "no_associated_account" if account ID not found,
+ *  - "unknown_error" if there's other error
+ *
+ */
+export const getCustomerAccountId = async (
+  phoneNumber: string,
+): Promise<string | GetAccountByPhoneErrorCode> => {
```

Comment:
I think returning string or error codes that are just strings is a little unspecific typing. If we want to return both error codes and string then I would either:
- Return `string` or `Error` (not errors for "normal" situations, like account not found).
- Type it up more specifically, maybe a return type like this: 
`{ number: string } | { errorCode: GetAccountByPhoneErrorCode }`

Reply:
Updated! No longer relevant since it now returns `string` or `undefined`